pulsar-client-cpp/include/pulsar/Client.h

```diff
@@ -42,110 +42,121 @@ class PulsarWrapper;
-     * @return ResultError if there was an error
-     */
-    Result createProducer(const std::string& topic, const ProducerConfiguration& conf,
-                          Producer& producer);
-
-    void createProducerAsync(const std::string& topic, CreateProducerCallback callback);
-
-    void createProducerAsync(const std::string& topic, ProducerConfiguration conf,
-                             CreateProducerCallback callback);
-
-    Result subscribe(const std::string& topic, const std::string& consumerName, Consumer& consumer);
-    Result subscribe(const std::string& topic, const std::string& consumerName,
-                     const ConsumerConfiguration& conf, Consumer& consumer);
-
-    void subscribeAsync(const std::string& topic, const std::string& consumerName,
-                        SubscribeCallback callback);
-    void subscribeAsync(const std::string& topic, const std::string& consumerName,
-                        const ConsumerConfiguration& conf, SubscribeCallback callback);
-
-    /**
-     * Create a topic reader with given {@code ReaderConfiguration} for reading messages from the specified topic.
-     * <p>
-     * The Reader provides a low-level abstraction that allows for manual positioning in the topic, without using a
-     * subscription. Reader can only work on non-partitioned topics.
-     * <p>
-     * The initial reader positioning is done by specifying a message id. The options are:
-     * <ul>
-     * <li><code>MessageId.earliest</code> : Start reading from the earliest message available in the topic
-     * <li><code>MessageId.latest</code> : Start reading from the end topic, only getting messages published after the
-     * reader was created
-     * <li><code>MessageId</code> : When passing a particular message id, the reader will position itself on that
-     * specific position. The first message to be read will be the message next to the specified messageId.
-     * </ul>
-     *
-     * @param topic
-     *            The name of the topic where to read
-     * @param startMessageId
-     *            The message id where the reader will position itself. The first message returned will be the one after
-     *            the specified startMessageId
-     * @param conf
-     *            The {@code ReaderConfiguration} object
-     * @return The {@code Reader} object
-     */
-     Result createReader(const std::string& topic, const MessageId& startMessageId,
-                       const ReaderConfiguration& conf, Reader& reader);
-
-     void createReaderAsync(const std::string& topic, const MessageId& startMessageId,
-                            const ReaderConfiguration& conf, ReaderCallback callback);
-
-    /**
-     *
-     * @return
-     */
-    Result close();
-
-    void closeAsync(CloseCallback callback);
-
-    void shutdown();
+  /**
+   * Create a Pulsar client object connecting to the specified cluster address and using
```

Comment:
For the same reason as indent width, we should probably increase `ColumnLimit` a little.